cpp/src/strings/json/json_path.cu

```diff
@@ -909,7 +909,6 @@ __launch_bounds__(block_size) __global__
   size_type tid    = threadIdx.x + (blockDim.x * blockIdx.x);
   size_type stride = blockDim.x * gridDim.x;
 
-  if (out_valid_count.has_value()) { *(out_valid_count.value()) = 0; }
```

Comment:
So this reset is no longer needed? I see that `out_valid_count` is then updated by `atomicAdd`, but what if `out_valid_count` was never been initialized before being passed into this function?

Reply:
It is initialized [here](https://github.com/rapidsai/cudf/blob/1aa4ec8775bdaff33eb69e5404506eb57beab10c/cpp/src/strings/json/json_path.cu#L1020) but I see your point. It can probably be safely initialized here, by `threadIdx.x = 0`, the thread that is responsible for loading the correct value.

Reply:
If so then this looks somewhat unsafe IMO because this needs to rely on the caller to initialize the variable without any guarantee from anywhere.

If we decided to not have the initialization here, it is better to have a comment line in the function doxygen clearly stressing/clarifying that.

Reply:
I moved the initialization @ttnghia @davidwendt please re-review.

Reply:
> If so then this looks somewhat unsafe IMO because this needs to rely on the caller to initialize the variable without any guarantee from anywhere.

Passing a pointer to an uninitialized device_scalar seems anti-RAII. If this were regular host code passing a pointer to a function and expecting that function to initialize it, without a very good reason (like that function has knowledge the caller can't know) I'd probably flag it as a code smell.